test/client/components/header.test.js

```diff
@@ -1,79 +1,74 @@
 import React from 'react';
-import { mount, shallow, render } from 'enzyme';
+import { mount, shallow } from 'enzyme';
 import { MemoryRouter } from 'react-router-dom';
 import renderer from 'react-test-renderer';
 import Header from '../../../public/src/components/header';
 
-const props = {
-  userId: 1,
-  userInfo: {
+describe('<Header />', () => {
+  // test variables
+  const testUser = {
     first_name: 'Mr.',
     last_name: 'Cat',
     email: 'mrcat@gmail.com',
     phone_number: '15555555555'
-  }
-};
-
-describe('<Header />', () => {
-  const wrapper = mount(
-    <MemoryRouter>
-      <Header />
-    </MemoryRouter>
-  );
-  it('renders correctly', () => {
-    const tree = renderer.create(
+  };
+  const props = {
+    userId: 1,
+    userInfo: testUser
+  };
+  describe('rendering', () => {
+    const wrapper = mount(
       <MemoryRouter>
         <Header />
       </MemoryRouter>
-    ).toJSON();
-    expect(tree).toMatchSnapshot();
-  });
-  it('should render <Navigation />', () => {
-    expect(wrapper.find('Navigation').length).toBe(1);
+    );
+    it('renders correctly', () => {
+      const tree = renderer.create(
+        <MemoryRouter>
+          <Header />
+        </MemoryRouter>
+      ).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+    it('should render <Navigation />', () => {
+      expect(wrapper.find('Navigation').length).toBe(1);
+    });
+    it('should render <Navbar />', () => {
+      expect(wrapper.find('Navbar').length).toBe(1);
+    });
+    it('should render <Navbar.Brand />', () => {
+      expect(wrapper.find('.navbar-brand').length).toBe(1);
+    });
   });
-  it('should render <Navbar />', () => {
-    expect(wrapper.find('Navbar').length).toBe(1);
+  describe('props', () => {
+    const wrapper = shallow(<Header />);
+    it('should receive a userInfo that is null',() => {
```

Comment:
Minor comment: maybe add more detail to the it string, specifying under what condition header should receive null userInfo.
This applies to the other null checks below.